dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/BNFHeadersImpl.java

```diff
@@ -245,6 +247,26 @@ public abstract class BNFHeadersImpl implements BNFHeaders, Externalizable {
      */
     private String forwardedPort = null;
 
+    /**
+     * Records whether a particular header has been added to this BNFHeaderImpl instance.
+     * Used to speed up getHeader calls for headers that have not been added, by avoiding
+     * unnecessary calls to findHeaders.
+     *
+     * Note: headers may be removed, after being added. In this case we do not reset the
+     * corresponding bit, to limit the implementation complexity. Therefore in cases where
+     * headers have been removed, we may get a false positive from headersAdded. This is OK
+     * because the next step is always findHeaders; we do not rely solely on headersAdded.
+     *
+     * Index: headerKey ordinal
+     * Value: true if header was added prior to checking the bit
+     */
+    private BitSet headersAdded = new BitSet();
+    /**
+     * The number of possible headers is virtually unlimited. We limit this fail-fast technique
+     * to the first N header keys registered in the system, to limit the complexity.
+     */
+    private final int maxHeadersAdded = 256;
```

Comment:
```suggestion
    private static final int maxHeadersAdded = 256;
```
No need to have this duplicated in each instance.

Reply:
yup - done